### PR TITLE
Add capture of static fields

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Capture static fields in dedicated section in snapshots. Static final fields are not captured.
Only static fields of the class of the instrumented methods are captured, not for every instance.

# Motivation

# Additional Notes

### CODE REVIEW

**Review by shatzi — APPROVED** [review]
LGTM

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/CapturedContextInstrumentor.java [hunk 1]
```diff
@@ -676,6 +679,59 @@ private void collectExceptionValue(AbstractInsnNode location, InsnList insnList)
     // stack: [throwable, capturedcontext]
   }
 
+  private void collectStaticFields(InsnList insnList) {
+    // count static fields
+    int fieldCount = 0;
+    for (FieldNode fieldNode : classNode.fields) {
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/CapturedContextInstrumentor.java:685]: true, it's a gap

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/CapturedContextInstrumentor.java [hunk 2]
```diff
@@ -676,6 +679,59 @@ private void collectExceptionValue(AbstractInsnNode location, InsnList insnList)
...
+    }
+    insnList.add(new InsnNode(Opcodes.DUP));
+    // stack: [capturedcontext, capturedcontext]
+    ldc(insnList, fieldCount);
+    // stack: [capturedcontext, capturedcontext, int]
+    insnList.add(new TypeInsnNode(Opcodes.ANEWARRAY, CAPTURED_VALUE.getInternalName()));
+    // stack: [capturedcontext, capturedcontext, array]
+    int counter = 0;
+    List<FieldNode> fieldList = new ArrayList<>(classNode.fields);
+    for (FieldNode fieldNode : fieldList) {
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/CapturedContextInstrumentor.java:702]: done

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/CapturedContextInstrumentor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/CapturedContextInstrumentor.java:685]: will address this in a next PR

**Pull request merged**